pandas/indexes/multi.py

```diff
@@ -116,12 +116,23 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
 
         return result
 
-    def _verify_integrity(self):
+    def _verify_integrity(self, new_labels=None, new_levels=None):
         """Raises ValueError if length of levels and labels don't match or any
-        label would exceed level bounds"""
+        label would exceed level bounds
+
+        Parameters
+        ----------
+        new_labels : optional list
+            Labels to check for validity. Defaults to current labels.
+        new_levels : optional list
+            Levels to check for validity. Defaults to current levels.
+        """
+
+
         # NOTE: Currently does not check, among other things, that cached
         # nlevels matches nor that sortorder matches actually sortorder.
-        labels, levels = self.labels, self.levels
+        labels = new_labels or self.labels
```

Comment:
I added a test for this, but didn't add a separate check within `_verify_integrity` because the check for list-like is already present in `set_levels` and `set_labels`: https://github.com/pydata/pandas/blob/master/pandas/indexes/multi.py#L221 and https://github.com/pydata/pandas/blob/master/pandas/indexes/multi.py#L322.